enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,410 @@
+
+When the management cluster is a standalone OCP cluster, there will also be a standard standalone USC instance in the respective platform namespace, managing an `UpdateStatus` that provides information about the management cluster itself.
+
+#### Standalone Clusters
+
+Standalone clusters are the primary target of the functionality and no special considerations are needed.
+
+#### Single-node Deployments or MicroShift
+
+USC and `UpdateStatus` API can work on SNO. USC is updated early in the procedure and will be able to expose the update status when the apiserver is available. USC and `UpdateStatus` can be disabled at installation time through the Capability mechanism for resource-sensitive SNO deployments. Existing SNO clusters using the `vCurrent` capability set would enable USC unless the cluster administrator intervenes by tweaking the capabilities before they update, which may warrant a release note when the feature goes GA.
+
+MicroShift is updated through `rpm-ostree` or `dnf`, so `UpdateStatus` API or any part of USC is not applicable and should not be included in MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+Full API proposal: https://github.com/openshift/api/pull/2012
+
+#### `UpdateStatus` API Overview
+
+```yaml
+apiVersion: update.openshift.io/v1alpha1
+kind: UpdateStatus
+metadata:
+  name: cluster
+spec: { }
+status:
+  controlPlane:
+    _: ...
+    informers:
+    - name: cvo-example-informer
+      insights: <list of insights reported by the informer>
+    - name: mco-example-informer
+      insights:
+      - type: ClusterVersion # CV status insight
+        _: ...
+      - type: ClusterOperator # CO status insight
+        _: ...
+      - type: UpdateHealth # General update health insight
+        _: ...
+    conditions: <list of standard kubernetes conditions>
+  workerPools:
+  - name: workers
+    _: ...
+    informers: <list of informers with reported insights>
+    conditions: <list of standard kubernetes conditions>
+  - name: infra
+    _: ...
+  conditions: <list of standard kubernetes conditions>    
+```
+
+The API has three conceptual layers:
+
+1. Through the innermost layer `.status.{controlPlane,workerPools[]}.informers`, the API exposes detailed information about individual concerns related to the update, called "Update Insights." The API is prepared to allow multiple external informers to contribute insights, but in this enhancement, the only informer is the USC itself.
```

Comment:
What would this API look like if the `informers` were a separate resource, that could be referenced, and then each actor could control one or more of these, and users could list and filter them as they need? It would greatly reduce the size of the output CR here right?

```
kind: UpdateInsight
spec:
  updatePool:
    type: Worker | ControlPlane
    worker:
      name: ... # we can have multiple worker pools?
  target:
    resource: Nodes | ClusterOperators | MachineConfigPools | ClusterVersions
    name: ....
status:
  conditions: []
  ... # I haven't worked out what else goes here
```